datahub-classify/src/datahub_classify/infotype_helper.py

```diff
@@ -186,7 +192,7 @@ def inspect_for_gender(
     try:
         if (
             prediction_factors_weights.get(NAME, 0) > 0
-            and debug_info.name == 1.0
+            and abs(1 - cast(float, debug_info.name)) < 1e-10
```

Comment:
should be using debug_info.name is not None to avoid the call to cast

Reply:
Modified the code